.ci/jobs/defaults.yml

```diff
@@ -0,0 +1,83 @@
+---
+
+##### GLOBAL METADATA
+
+- meta:
+    cluster: elasticsearch-ci
+
+##### JOB DEFAULTS
+
+- job:
+    vault: []
+    node: "rally"
+    concurrent: true
+    logrotate:
+      daysToKeep: 30
+      numToKeep: 500
+      artifactDaysToKeep: 7
+    parameters:
+      - string:
+          name: branch_specifier
+          default: "refs/heads/{branch}"
+          description: "the Git branch specifier to build (&lt;branchName&gt;, &lt;tagName&gt;, &lt;commitId&gt;, etc.)\n"
+    scm:
+      - git:
+          name: origin
+          # master node jenkins user ~/.ssh
+          credentials-id: f6c7695a-671e-4f4f-a331-acdce44ff9ba
+          reference-repo: "/var/lib/jenkins/.git-references/rally.git"
+          branches:
+            - "${branch_specifier}"
+          url: "https://github.com/elastic/rally.git"
+          basedir: ""
+          wipe-workspace: true
+    triggers: []
+    wrappers:
+      - ansicolor
+      - timeout:
+          type: absolute
+          timeout: 80
+          fail: true
+      - timestamps
+    properties:
+      - github:
+          url: https://github.com/elastic/rally/
+      - inject:
+          properties-content: |
+            HOME=$JENKINS_HOME
+    builders:
+      - inject:
+          properties-content: |
+            HOME=$JENKINS_HOME
+            JAVA7_HOME=$HOME/.java/java7
+            JAVA8_HOME=$HOME/.java/java8
+            JAVA9_HOME=$HOME/.java/java9
+            JAVA10_HOME=$HOME/.java/java10
+            JAVA11_HOME=$HOME/.java/java11
+            JAVA12_HOME=$HOME/.java/openjdk12
+            JAVA13_HOME=$HOME/.java/openjdk13
+            JAVA14_HOME=$HOME/.java/openjdk14
+            JAVA15_HOME=$HOME/.java/openjdk15
+            JAVA16_HOME=$HOME/.java/openjdk16
+            JAVA17_HOME=$HOME/.java/openjdk17
+            ES_JAVA_HOME=$JAVA17_HOME
+            RALLY_HOME=$WORKSPACE
+      - shell: |
+          #!/usr/local/bin/runbld
+          bash .ci/build.sh build
```

Comment:
We've faced this issue elsewhere, we need to prefix with [set -o errexit](https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin) because it's possible that tests failed via `build.sh` but then CI reports success due to the `archive` function below returning true.